### PR TITLE
BigInt.prototype.toString

### PR DESCRIPTION
**Pull request by cxielarko**
This is an initial set of tests for BigInt.prototype.toString based on existing tests for Number.prototype.toString.

### CODE REVIEW

File: test/built-ins/BigInt/prototype/toString/length.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  17 ECMAScript Standard Built-in Objects
+includes: [propertyHelper.js]
+---*/
+
+verifyProperty(BigInt.prototype.toString, "length", {
+  value: 1,
+  writable: false,
+  enumerable: false,
+  configurable: true
+});
```
Comment (rwaldron) [inline test/built-ins/BigInt/prototype/toString/length.js]: 👍

File: test/built-ins/BigInt/prototype/toString/length.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-bigint.prototype.tostring
+description: BigInt.prototype.toString.length property descriptor
+info: >
+  BigInt.prototype.toString ( [ radix ] )
+
+  17 ECMAScript Standard Built-in Objects
+includes: [propertyHelper.js]
```
Comment (rwaldron) [inline test/built-ins/BigInt/prototype/toString/length.js]: `features: [BigInt]`

File: test/built-ins/BigInt/prototype/toString/name.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-bigint.prototype.tostring
+description: BigInt.prototype.toString.name property descriptor
+info: >
+  BigInt.prototype.toString ( [ radix ] )
+
+  17 ECMAScript Standard Built-in Objects
+includes: [propertyHelper.js]
```
Comment (rwaldron) [inline test/built-ins/BigInt/prototype/toString/name.js]: `features: [BigInt]`

File: test/built-ins/BigInt/prototype/toString/prop-desc.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-bigint.prototype.tostring
+description: BigInt.prototype.toString property descriptor
+info: >
+  BigInt.prototype.toString ( [ radix ] )
+
+  17 ECMAScript Standard Built-in Objects
+includes: [propertyHelper.js]
```
Comment (rwaldron) [inline test/built-ins/BigInt/prototype/toString/prop-desc.js]: `features: [BigInt]`

File: test/built-ins/BigInt/prototype/toString/prototype-call.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+description: Direct toString on BigInt prototype
+info: >
+  BigInt.prototype.toString ( [ radix ] )
+
+  Let x be ? thisBigIntValue(this value).
+
+  Properties of the BigInt Prototype Object
+
+  The BigInt prototype is not a BigInt object; it does not have a
+  [[BigIntData]] internal slot.
```
Comment (rwaldron) [inline test/built-ins/BigInt/prototype/toString/prototype-call.js]: `features: [BigInt]`

File: test/built-ins/BigInt/prototype/toString/radix-2-to-36.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+info: >
+  BigInt.prototype.toString ( [ radix ] )
+
+  [...]
+  6. If radixNumber = 10, return ! ToString(x).
+  7. Return the String representation of this Number value using the
+     radix specified by radixNumber. Letters a-z are used for digits
+     with values 10 through 35. The precise algorithm is
+     implementation-dependent, however the algorithm should be a
+     generalization of that specified in 3.1.4.1.
```
Comment (rwaldron) [inline test/built-ins/BigInt/prototype/toString/radix-2-to-36.js]:
```
features: [BigInt, arrow-function]
```

File: test/built-ins/BigInt/prototype/toString/radix-err.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+esid: sec-bigint.prototype.tostring
+description: toString with invalid radix
+info: >
+  BigInt.prototype.toString ( [ radix ] )
+
+  [...]
+  4. Else, let radixNumber be ? ToInteger(radix).
+  5. If radixNumber < 2 or radixNumber > 36, throw a RangeError
+     exception.
+---*/
```
Comment (rwaldron) [inline test/built-ins/BigInt/prototype/toString/radix-err.js]:
```
features: [BigInt, arrow-function]
```

File: test/built-ins/BigInt/prototype/toString/prototype-call.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+---*/
+
+assert.throws(TypeError, () => BigInt.prototype.toString());
+assert.throws(TypeError, () => BigInt.prototype.toString(10));
+assert.throws(TypeError, () => BigInt.prototype.toString(undefined));
```
Comment (rwaldron) [inline test/built-ins/BigInt/prototype/toString/prototype-call.js]: Did you mean to write `BigInt.prototype.toString.call()` for all of these?

> The BigInt prototype is not a BigInt object; it does not have a [[BigIntData]] internal slot.

File: test/built-ins/BigInt/prototype/toString/radix-2-to-36.js [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+---*/
+
+for (let r = 2; r <= 36; r++) {
+  assert.throws(TypeError, () => BigInt.prototype.toString(r));
```
Comment (rwaldron) [inline test/built-ins/BigInt/prototype/toString/radix-2-to-36.js]: Same question as above, re: 

> The BigInt prototype is not a BigInt object; it does not have a [[BigIntData]] internal slot.

File: test/built-ins/BigInt/prototype/toString/radix-err.js [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+
+for (let r of [0, 1, 37, null]) {
+  assert.throws(TypeError, () => BigInt.prototype.toString(r));
```
Comment (rwaldron) [inline test/built-ins/BigInt/prototype/toString/radix-err.js]: Same question as above, re: 

> The BigInt prototype is not a BigInt object; it does not have a [[BigIntData]] internal slot.

File: test/built-ins/BigInt/prototype/toString/non-bigint.js [hunk 11]
```diff
@@ -0,0 +1,16 @@
+// Copyright 2017 Robin Templeton. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-bigint.prototype.tostring
+description: Transferring toString to non-BigInt objects
+info: >
+  BigInt.prototype.toString ( [ radix ] )
+
+  1. Let x be ? thisBigIntValue(this value).
```
Comment (rwaldron) [inline test/built-ins/BigInt/prototype/toString/non-bigint.js]:
```
features: [BigInt, arrow-function]
```

**Comment by leobalter** [comment]
ping.

**Comment by leobalter** [comment]
ping

**Comment by leobalter** [comment]
at this point I might apply the fixes on top of this work if time allows today.

**Comment by leobalter** [comment]
superseeded by #1272

**Pull request closed without merging**